### PR TITLE
New test cast - Ephemeral storage

### PR DESCRIPTION
**Pull request by AviadP**
Create test case and infrastructure for ephemeral storage

### CODE REVIEW

**Review by OdedViner — DISMISSED** [review]

File: ocs_ci/ocs/ephernal_storage.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                    "image": "busybox:1.28",
+                    "volumeMounts": [
+                        {"mountPath": "/scratch", "name": "scratch-volume"}
+                    ],
+                    "command": ["sleep", "1000000"],
+                }
+            ],
+            "volumes": [],
+        },
+    }
```
Comment (prsurve) [inline ocs_ci/ocs/ephernal_storage.py:42]: why are we writing pod template here it should be part of yaml inside ocs_ci/templates/app-pods

File: ocs_ci/ocs/ephernal_storage.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        "apiVersion": "v1",
+        "metadata": {
+            "name": "pod_name1",
+            "namespace": config.ENV_DATA["cluster_namespace"],
+        },
+        "spec": {
+            "containers": [
+                {
+                    "name": "my-frontend",
+                    "image": "busybox:1.28",
```
Comment (prsurve) [inline ocs_ci/ocs/ephernal_storage.py:33]: we cannot use busybox image from docker.io we will get rate limit issue

**Review by prsurve — DISMISSED** [review]

File: ocs_ci/ocs/ephernal_storage.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import Pod
+from ocs_ci.utility.templating import load_yaml
+
+log = getLogger(__name__)
+
+
+class EphernalPodFactory:
```
Comment (ebenahar) [inline ocs_ci/ocs/ephernal_storage.py:18]: EphemeralPodFactory:

File: ocs_ci/ocs/ephernal_storage.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        return pod_dict
+
+    @staticmethod
+    def create_rbd_pod_dict(pod_name: str, **kwargs) -> dict:
+        pod_dict_path = EPHEMERAL_RBD_POD_YAML
+        pod_dict = load_yaml(pod_dict_path)
+        return pod_dict
+
+    @staticmethod
+    def create_ephmeral_pod(pod_name: str, storage_type: str, **kwargs) -> Pod:
```
Comment (ebenahar) [inline ocs_ci/ocs/ephernal_storage.py:40]: create_ephemeral_pod

File: tests/functional/pod_and_daemons/test_ephemeral_pod.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    tier1,
+    brown_squad,
+)
+
+log = getLogger(__name__)
+
+
+@tier1
+@brown_squad
+class TestEphernalPod:
```
Comment (ebenahar) [inline tests/functional/pod_and_daemons/test_ephemeral_pod.py:22]: TestEphemeralPod

File: tests/functional/pod_and_daemons/test_ephemeral_pod.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    @pytest.mark.parametrize(
+        argnames=["interface"], argvalues=[[CEPHFS_INTERFACE], [RBD_INTERFACE]]
+    )
+    def test_ephernal_pod_creation(self, interface) -> None:
```
Comment (ebenahar) [inline tests/functional/pod_and_daemons/test_ephemeral_pod.py:26]: test_ephemeral_pod_creation

File: tests/functional/pod_and_daemons/test_ephemeral_pod.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import (
+    tier1,
+    brown_squad,
+)
+
+log = getLogger(__name__)
+
+
+@tier1
```
Comment (ebenahar) [inline tests/functional/pod_and_daemons/test_ephemeral_pod.py:20]: missing polarion ID

**Review by OdedViner — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**